### PR TITLE
{EventHub} Add Breaking change message for `az eventhubs namespace network-rule` cmdlets.

### PR DESCRIPTION
**Pull request by schaudhari6254888**
**Related command**

**Description**
We will change the format of this command in Future release. the parameter `vnet-name` will be deprecated in future.
We will offer simultaneous addition and removal of multiple IP rules and subnet rules with the new future release cmdlets.
We are targeting these breaking change in this May breaking change cycle.

We will Replace the following command.
1. `az eventhubs namespace network-rule` with `az eventhubs namespace network-rule-set`
2. `az eventhubs namespace network-rule add` with `az eventhubs namespace network-rule-set ip-rule add`, `az eventhubs namespace network-rule-set virtual-network-rule add`
3. `az eventhubs namespace network-rule remove` with `az eventhubs namespace network-rule-set ip-rule remove`, `az eventhubs namespace network-rule-set virtual-network-rule remove`
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
EventHub

**Comment by schaudhari6254888** [comment]
This PR is about breaking change Alert Notification for EventHub Network-Rule-Set cmdlets.
I have mentioned the changes in the description. Can someone review & merge this PR.
Any Suggestions are Welcome.
Thank you in advance!

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def cli_networkrule_createupdate(cmd, client, resource_group_name, namespace_nam
     NWRuleSetIpRules = cmd.get_models('NWRuleSetIpRules', resource_type=ResourceType.MGMT_EVENTHUB)
 
     netwrokruleset = client.get_network_rule_set(resource_group_name, namespace_name)
-
+    logger.warning('This version will be depracated & latest version will release in breaking change release.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:308]: Please leverage `deprecate_info`

Comment (schaudhari6254888) [reply src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:308]: Hey @evelyn-ys , I have added the depreciation_info, Can you please check now.

Also for you reference I am attaching PR Link which is in draft, that will show ho will be the future cmdlets for network-rule.
https://github.com/Azure/azure-cli/pull/25792

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 2]
```diff
@@ -177,9 +177,9 @@ def load_arguments_eh(self, _):
     for scope in ['eventhubs namespace network-rule add', 'eventhubs namespace network-rule remove']:
         with self.argument_context(scope, resource_type=ResourceType.MGMT_EVENTHUB, min_api='2017-04-01') as c:
             c.argument('subnet', arg_group='Virtual Network Rule', options_list=['--subnet'], help='Name or ID of subnet. If name is supplied, `--vnet-name` must be supplied.')
-            c.argument('ip_mask', arg_group='IP Address Rule', options_list=['--ip-address'], help='IPv4 address or CIDR range - 10.6.0.0/24')
+            c.argument('ip_mask', arg_group='IP Address Rule', options_list=['--ip-address'], help='IPv4 address or CIDR range - 10.6.0.0/24', deprecate_info=c.deprecate(redirect='--ip-rule', expiration='2.50.0'))
```
[the remark this answers is not in the document]
Comment (schaudhari6254888) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:180]: Yes

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**